flink-runtime/src/main/java/org/apache/flink/runtime/state/TaskExecutorLocalStateStoresManager.java

```diff
@@ -186,6 +210,34 @@ private void doRelease(Iterable<TaskLocalStateStore> toRelease) {
 		return localRecoveryMode;
 	}
 
+	private File[] allocationBaseDirectories(AllocationID allocationID) {
+		File[] allocationDirectories = new File[localStateRootDirectories.length];
+		for (int i = 0; i < localStateRootDirectories.length; ++i) {
+			allocationDirectories[i] = new File(localStateRootDirectories[i], allocationSubDirString(allocationID));
```

Comment:
`allocationSubDirString` could be pulled out of the loop.

Reply:
👍